lib/features/home_screen/search_page.dart

```diff
@@ -1,13 +1,190 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_modular/flutter_modular.dart';
+import 'package:marvel_app/features/home_screen/cubits/search_cubit/search_cubit.dart';
+import 'package:marvel_app/features/home_screen/widgets/list_item_widget.dart';
+import 'package:marvel_app/features/home_screen/widgets/loading_widget.dart';
+import 'package:marvel_app/resources/dimens/search_page_dimens.dart';
+import 'package:marvel_app/resources/strings/search_page_strings.dart';
 
-class SearchPage extends StatefulWidget{
+import '../../core/models/comics_data_wrapper.dart';
+import 'cubits/search_cubit/search_state.dart';
+
+class SearchPage extends StatefulWidget {
   @override
   _SearchPageState createState() => _SearchPageState();
 }
 
-class _SearchPageState extends State<SearchPage>{
+class _SearchPageState extends State<SearchPage> {
+  SearchCubit searchCubit = Modular.get<SearchCubit>();
+  
+  bool visibleText = false;
+  Color color = const Color.fromRGBO(169, 169, 169, 1);
+  TextEditingController controller = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    controller.addListener(() {
+      if(controller.text.isNotEmpty) {
+        setState(() {
+          color = const Color.fromRGBO(100, 100, 100, 1);
+          visibleText = true;
+        });
+      } else {
+        setState(() {
+          color = const Color.fromRGBO(169, 169, 169, 1);
```

Comment:
Better move that logic to place where you use it and make something like that:
`color: visibleText ? color1 : color2`